src/test/java/org/kiwiproject/dropwizard/util/health/keystore/ExpiringKeystoreHealthCheckTest.java

```diff
@@ -0,0 +1,267 @@
+            );
+
+            List<BasicCertInfo> validCerts = getBasicCertInfoList(details, "validCerts");
+            softly.assertThat(validCerts).isEmpty();
+
+            List<BasicCertInfo> expiredCerts = getBasicCertInfoList(details, "expiredCerts");
+            softly.assertThat(expiredCerts).isEmpty();
+
+            List<BasicCertInfo> expiringCerts = getBasicCertInfoList(details, "expiringCerts");
+            softly.assertThat(expiringCerts).hasSize(1);
+
+            var certInfo = first(expiringCerts);
+            softly.assertThat(certInfo.getSubjectDN()).isEqualTo("CN=Unit Test, OU=Development, O=Project, L=Here, ST=VA, C=US");
+            softly.assertThat(certInfo.getIssuerDN()).isEqualTo("CN=Unit Test, OU=Development, O=Project, L=Here, ST=VA, C=US");
+
+            var sixtyDaysAgo = ZonedDateTime.now(ZoneId.of("UTC")).minusDays(60);
+            var sixtyDaysAgoString = DATE_FORMATTER.format(sixtyDaysAgo);
+            softly.assertThat(certInfo.getIssuedOn()).startsWith(sixtyDaysAgoString);
+
+            var expirationDate = sixtyDaysAgo.plusDays(75);
+            var expirationDateString = DATE_FORMATTER.format(expirationDate);
+            softly.assertThat(certInfo.getExpiresOn()).startsWith(expirationDateString);
+
+        }
+
+        @SuppressWarnings("unchecked")
+        @Test
+        void whenAnExceptionIsThrown(SoftAssertions softly, @TempDir Path tempDir) throws Exception {
+            var dn = "CN=Unit Test, OU=Development, O=Project, L=Here, ST=VA, C=US";
+            var path = createTemporaryKeystore(tempDir, 75, dn);
+
+            var config = KeystoreConfig.builder()
+                    .name("Test Keystore Config")
+                    .path(path)
+                    .pass(" not-the-password")
+                    .type(KeyStoreType.PKCS12.name())
+                    .build();
+
+            var healthCheck = new ExpiringKeystoreHealthCheck(config);
+            var result = healthCheck.check();
+
+            softly.assertThat(result.isHealthy()).isFalse();
+            softly.assertThat(result.getMessage()).isEqualTo(path + ": exception");
+            softly.assertThat(result.getError()).isNotNull();
+
+            var details = result.getDetails();
+            softly.assertThat(details).contains(
+                    entry("path", path),
+                    entry("ttl", Duration.days(30)) // default ttl
+            );
+
+            List<BasicCertInfo> validCerts = getBasicCertInfoList(details, "validCerts");
+            softly.assertThat(validCerts).isEmpty();
+
+            List<BasicCertInfo> expiredCerts = getBasicCertInfoList(details, "expiredCerts");
+            softly.assertThat(expiredCerts).isEmpty();
+
+            List<BasicCertInfo> expiringCerts = getBasicCertInfoList(details, "expiringCerts");
+            softly.assertThat(expiringCerts).isEmpty();
+        }
+    }
+
+    /**
+     * The keytool command that generated the valid cert:
```

Comment:
Is all this javadoc now incorrect, since you changed the original code to generate certs on the fly?

Reply:
Yes, removed